core/executor.py

```diff
@@ -247,7 +248,13 @@ def RunSimpleCommand(self, cmd_val, cmd_st, do_fork, call_procs=True):
     if do_fork:
       thunk = process.ExternalThunk(self.ext_prog, argv0_path, cmd_val, environ)
       p = process.Process(thunk, self.job_state, self.tracer)
-      status = p.RunWait(self.waiter, trace.External(cmd_val.argv))
+      if pipeline is not None:
```

Comment:
OK I think I see why you did this, let me think about whether there is a structure we can use that avoids threading this `parent_pipeline` param through a lot of places ...